js/components/dateRange.js

```diff
@@ -0,0 +1,280 @@
+    value: function(val) {
+      if (!val && ['default', 'always'].indexOf(this.autofill) > -1 && (this.required || this.autofill === 'always')) {
+        this.value = {
+          start: this.today,
+          end: this.today
+        };
+
+        if (this.dateRange) {
+          this.dateRange.set(val, true);
+        }
+
+        this.$emit('_input', this.name, val, false, true);
+
+        return;
+      }
+
+      if (this.isPreview && this.$v.value.$invalid) {
+        this.highlightError();
+      }
+
+      if (this.dateRange) {
+        this.dateRange.set(val, true);
+      }
+
+      this.$emit('_input', this.name, val, false, true);
+    },
+    selectedRange: function(val) {
+      var newDate = {
+        start: '',
+        end: ''
+      };
+
+      switch (val) {
+        case T('widgets.form.dateRange.predefinedRanges.today'):
+          newDate = this.getDate('today');
+          break;
+        case T('widgets.form.dateRange.predefinedRanges.yesterday'):
+          newDate = this.getDate('yesterday');
+          break;
+        case T('widgets.form.dateRange.predefinedRanges.tomorrow'):
+          newDate = this.getDate('tomorrow');
+          break;
+        case T('widgets.form.dateRange.predefinedRanges.nextWeek'):
+          newDate = this.getDate('nextWeek');
+          break;
+        case T('widgets.form.dateRange.predefinedRanges.lastWeek'):
+          newDate = this.getDate('lastWeek');
+          break;
+        case T('widgets.form.dateRange.predefinedRanges.nextMonth'):
+          newDate = this.getDate('nextMonth');
+          break;
+        case T('widgets.form.dateRange.predefinedRanges.lastMonth'):
+          newDate = this.getDate('lastMonth');
+          break;
+        default:
+          newDate = this.getDate('today');
+          break;
+      }
+
+      if (this.dateRange) {
+        this.dateRange.set(newDate, true);
+      }
+
+      this.value = newDate;
+      this.$emit('_input', this.name, this.value, false, true);
+    }
+  },
+  created: function() {
+    Fliplet.Hooks.on('beforeFormSubmit', this.onBeforeSubmit);
+  },
+  destroyed: function() {
+    Fliplet.Hooks.off('beforeFormSubmit', this.onBeforeSubmit);
+  },
+  methods: {
+    initDaterange: function() {
+      var $vm = this;
+
+      if (this.dateRange && !this.$refs.dateRange) {
+        return;
+      }
+
+      if (!this.value.start || !this.value.end && ['default', 'always'].indexOf(this.autofill) > -1 && (this.required || this.autofill === 'always')) {
+        $vm.value = {
+          start: this.today,
+          end: this.today
+        };
+      }
+
+      this.dateRange = Fliplet.UI.DateRange(this.$refs.dateRange, {
+        required: this.required || this.autofill === 'always',
+        forceRequire: false,
+        value: this.value,
+        readonly: this.readonly
+      });
+
+      this.dateRange.change(function(value) {
+        if (!value || !value.start || !value.end) {
+          this.selectedRange = '';
```

Comment:
`this.selectedRange` should be set to `''` whenever the value changes after the range is moved away from a predefined range.